### PR TITLE
feat(phoenix): Add Room model

### PR DESCRIPTION
**Pull request by mrapacz**
**Summary**
This PR adds Room model along with RoomSubject which is a join table between rooms and subjects (we want each subject to be in many rooms and each room to possibly aggregate any subjects)

Below the way the generated Room table looks:
image

And RoomSubject:
image

And finally updated Subject:
image

**Related issues**
#34 

**Test plan**
`mix ecto.migrate && mix test`

### CODE REVIEW

File: aion/web/models/room.ex [hunk 1]
```diff
@@ -0,0 +1,23 @@
+defmodule Aion.Room do
+  use Aion.Web, :model
+  alias Aion.{Subject, RoomSubject, User}
+
+  schema "rooms" do
+    field :name, :string
+    field :description, :string
+    many_to_many :subjects, Subject, join_through: RoomSubject
```
Comment (jtkpiotr) [inline aion/web/models/room.ex]: Don't we need to specify on_delete here?

File: aion/web/models/subject.ex [hunk 2]
```diff
@@ -1,9 +1,9 @@
 defmodule Aion.Subject do
   use Aion.Web, :model
-
+  alias Aion.{Room, RoomSubject}
   schema "subjects" do
     field :name, :string
-
+    many_to_many :rooms, Room, join_through: RoomSubject
```
Comment (jtkpiotr) [inline aion/web/models/subject.ex]: The same as above

**Comment by mrapacz** [comment]
@jtkpiotr correct, fixed. Will take care of fixing tests etc. soon

**Comment by mrapacz** [comment]
@jtkpiotr Fixed all of this, you can check it out once again

**Review by jtkpiotr — APPROVED** [review]

File: aion/test/models/room_subject_test.exs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+defmodule Aion.RoomSubjectTest do
+  use Aion.ModelCase
+
+  alias Aion.RoomSubject
+
+  @valid_attrs %{}
+
+  test "changeset with valid attributes" do
```
Comment (pmrukot) [inline aion/test/models/room_subject_test.exs:8]: This test is always true since `@valid_attrs %{}`

File: aion/test/models/room_test.exs [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+  @invalid_attrs %{}
+
+  test "changeset with valid attributes" do
+    changeset = Room.changeset(%Room{}, @valid_attrs)
+    assert changeset.valid?
+  end
+
+  test "changeset with invalid attributes" do
+    changeset = Room.changeset(%Room{}, @invalid_attrs)
+    refute changeset.valid?
```
Comment (pmrukot) [inline aion/test/models/room_test.exs:16]: This one too.

**Review by pmrukot — APPROVED** [review]

File: aion/test/models/room_test.exs (hunk 4 above)
Comment (mrapacz) [reply aion/test/models/room_test.exs:16]: That's good, we may change the model at some point to a model that would make this test fail, it's good to get an alert that things that we say 'are always true' are actually not after a change.

**Pull request merged**